packages/gatsby/src/redux/reducers/__tests__/redirects.js

```diff
@@ -0,0 +1,126 @@
+const reducer = require(`../redirects`)
+
+describe(`redirects`, () => {
+  it(`lets you redirect to an internal url`, () => {
+    const action = {
+      type: `CREATE_REDIRECT`,
+      payload: {
+        fromPath: `/page1`,
+        toPath: `/page1/`,
+      },
+    }
+
+    let state = reducer(undefined, action)
+
+    expect(state).toEqual(
+      [
+        {
+          fromPath: `/page1`,
+          toPath: `/page1/`,
+        },
+      ]
+    )
+  })
+
+  it(`lets you redirect to an external url`, () => {
+    const action = {
+      type: `CREATE_REDIRECT`,
+      payload: {
+        fromPath: `/page1`,
+        toPath: `https://example.com`,
+      },
+    }
+
+    let state = reducer(undefined, action)
+
+    expect(state).toEqual(
+      [
+        {
+          fromPath: `/page1`,
+          toPath: `https://example.com`,
+        },
+      ]
+    )
+  })
+
+  it(`lets you redirect using any protocol`, () => {
```

Comment:
Might be valuable to refactor this slightly so that each protocol is tested separately? It's also easier to refactor at some later date, i.e. we don't need to be picky about order of actions.

So instead of this test, it'd be separate ones, e.g.

```js
it(`lets you redirect using https`, () => {
  const action = {
    type: `CREATE_REDIRECT`,
    payload: {
      fromPath: `/page1`,
      toPath: `https://example.com`,
    },
  }

  const state = reducer(undefined, action)

  expect(state).toEqual([
    {
      fromPath: `/page1`,
      toPath: `https://example.com`
    }
  ])
});
```

or you could get crazy and do something like this :)

```js
;[
  [`https`, `https://example.com`],
  [`http`, `https://example.com`] // and the other ones too
]
  .forEach(([protocol, toPath], index) => {
    it(`lets you redirect using ${protocol}`, () => {
      const fromPath = `/page${index}`
      const action = {
        type: `CREATE_REDIRECT`,
        payload: {
          fromPath,
          toPath
        }
      }

      expect(reducer(undefined, action)).toEqual([
        {
          fromPath,
          toPath
        }
      ])
    })
  })
```

Reply:
This is super helpful! I like how clean the last example is so I'm going to roll with that :)